fast_response/FastResponseAnalysis.py

```diff
@@ -992,6 +1020,424 @@ def unblind_TS(self, custom_events=None):
+        # Compute the entire likelihood space P(X | RA, DEC, ns, gamma)
+        # Apply Bayes rule, marginalize over nuisance parameters to get P(RA, DEC | X)
+ 
+        #flux/ns to test: given as list  
+        #Give prior as a function f(ns, gamma, ra, dec)
+            #As currently implemented,the llh will have gamma must be constant set by the self.index
+            #Prior flat in ns and spatially by default. 
+
+        #plotting_location is the directory where plots should be output
+        #custom events will be injected
+        #useFlux determines whether fluxToTest is treated as ns or flux
+
+
+        t1 = time.time()
+        val=None
+        skymap = self.skymap
+        
+        
+        spatial_prior = SpatialPrior(
+            skymap, 
+            containment = self._containment,allow_neg=self._allow_neg)
+        
+        nside=self.nside
+        decs_list = hp.pix2ang(nside, np.arange(hp.nside2npix(nside)))
+
+
+        test_flux=1e-17
+        if(useFlux):
+            mean_flux_arr=fluxToTest
+            nsPerFluxArr=[self.fluxToNs_decDependent(test_flux,d)/test_flux for d in decs_list]
+        else:
+            #work in uniform ns
+            mean_flux_arr=fluxToTest
+            nsPerFluxArr=[1]*12*pow(nside,2) 
+
+
+        for fluxT in mean_flux_arr:
+            temp_val = self.llh.scan(0.0,0.0, scramble = False,spatial_prior=spatial_prior,
+                    time_mask = [self.duration/2., self.centertime],
+                    pixel_scan=[nside, self._pixel_scan_nsigma],
+                    fixed=['nsignal'],
+                    nsignal=np.array([fluxT*conversion for conversion in nsPerFluxArr],dtype=float), 
+
+                    inject=custom_events
+                    )
+            if val is None:
+                val=temp_val
+            else:
+                val=np.hstack((val,temp_val))
+        ras=val["ra"]
+        decs=val["dec"]
+        nss=val["nsignal"]
+    
+        #Placeholder: not required in fixed index analyses, but future improvements could expand this
+        if "gamma" in val.dtype.names:
+            gammas=val["gamma"]
+        
+        TS_spatial_prior_0=val["TS_spatial_prior_0"]
+        idx_max=np.argmax(TS_spatial_prior_0)
+        ramax,decmax=ras[idx_max],decs[idx_max]
+
+        logProbs=TS_spatial_prior_0/2 
+        finalProbs={}
+
+
+        #for each pixel we have lists [ns_1,ns_2,...ns_i], [logP_1,logP_2,...,logP_3]
+        #we want to integrate over ns
+
+        pixel_list_full=list(zip(ras,decs))
+        for pix in set(pixel_list_full):
+            idxes=np.where((ras==pix[0])&(decs==pix[1]))[0]
+            sort_idx=np.argsort(nss[idxes])
+            idxes=idxes[sort_idx]
+            nss_temp=nss[idxes]
+            probs_temp=logProbs[idxes]
+            ns_diff=np.diff(nss_temp)
+            
+            finalProbs[pix]=-np.inf
+            if(len(idxes)!=len(fluxToTest)):
+                print("Not all fluxes have probabilities")
```

Comment:
can you raise an error here? just make it clearer that the code is exiting and this part failed because of this condition